README.md

```diff
@@ -0,0 +1,191 @@
+# Xmartlabs' Android Style Guide
+
+This is the Xmartlabs' Android and Java Styles Guide. These are conventions in general followed by the team.
+
+## Java
+
+The idea is to follow [Google Java Style](http://google.github.io/styleguide/javaguide.html), but with some changes and adding ideas.
+
+### Changes
+
+#### Vertical whitespace
+
+There must be no vertical whitespace before the first member and after the last one of a class.
+
+```java
+public class ClassExample {
+  private int firstMember; // no vertical whitespace before this line
+
+  // ...
+
+  public void lastMember() {
+    // ...
+  } // no vertical whitespace after this line
+}
+```
+
+Multiple vertical whitespaces must never be used.
+
+#### Array initializers
+
+The preferred array initializer "block-like construct" style when it doesn't fit in one line is one value per line:
+
+```java
+new int[] {
+  0,
+  1,
+  2,
+  3,
+}
+```
+
+#### Numeric literals
+
+`float`-valued literals use a lowercase `f` suffix: `100.23f`.
+
+#### Class and interface names
+
+If it's a list of an entity, use singular: `CustomerListFragment` for instance.
+
+#### Field, method, variable and parameter names
+
+If you need to name a `List` of activities for example, just name it `activities` and if it's `ListView`: `activitiesListView`. So we use plural here.
+
+#### Local variable names
+
+One-character names are only permitted if they are `i`, `j` or `k` indexes in loops. They must be used in this respective order when nesting them. Example:
```

Comment:
Ok I understand your point. So what do you think about adding some kind of
clause like "are not allowed at least..." or "are allowed in some necessary
case but it is not recommended"?


[image: xmartlabs website] <http://xmartlabs.com/>
MIGUEL REVETRIA
iOS Engineer
_________________
<https://www.linkedin.com/in/mrevetria> <https://github.com/m-revetria>

On Fri, Jan 22, 2016 at 2:06 PM, Santiago Castro <notifications@github.com>
wrote:

> In README.md
> <https://github.com/xmartlabs/Android-Style-Guide/pull/1#discussion_r50562319>
> :
>
> > +
> > +#### Numeric literals
> > +
> > +`float`-valued literals use a lowercase `f` suffix: `100.23f`.
> > +
> > +#### Class and interface names
> > +
> > +If it's a list of an entity, use singular: `CustomerListFragment` for instance.
> > +
> > +#### Field, method, variable and parameter names
> > +
> > +If you need to name a `List` of activities for example, just name it `activities` and if it's `ListView`: `activitiesListView`. So we use plural here.
> > +
> > +#### Local variable names
> > +
> > +One-character names are only permitted if they are `i`, `j` or `k` indexes in loops. They must be used in this respective order when nesting them. Example:
>
> I agree on the itemFound thing. But there are some cases in which you
> could want to continue iterating after the loop
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xmartlabs/Android-Style-Guide/pull/1/files#r50562319>.
>